src/expr/src/scalar/func/impls/map.rs

```diff
@@ -125,3 +126,76 @@ impl fmt::Display for MapLength {
         f.write_str("map_length")
     }
 }
+
+#[derive(
+    Arbitrary, Ord, PartialOrd, Clone, Debug, Eq, PartialEq, Serialize, Deserialize, Hash, MzReflect,
+)]
+pub struct MapBuildFromRecordList {
+    pub value_type: ScalarType,
+}
+
+impl LazyUnaryFunc for MapBuildFromRecordList {
+    fn eval<'a>(
+        &'a self,
+        datums: &[Datum<'a>],
+        temp_storage: &'a RowArena,
+        a: &'a MirScalarExpr,
+    ) -> Result<Datum<'a>, EvalError> {
+        let a = a.eval(datums, temp_storage)?;
+        if a.is_null() {
+            return Ok(Datum::Null);
+        }
+        let list = a.unwrap_list();
+        let mut map = std::collections::BTreeMap::new();
+
+        for i in list.iter() {
+            if i.is_null() {
+                continue;
+            }
+
+            for (k, v) in i.unwrap_list().iter().tuples() {
+                if k.is_null() {
+                    continue;
+                }
+                map.insert(k.unwrap_str(), v);
```

Comment:
Just confirming my understanding, unlike `map_agg`, `map_build` doesn't provide any guarantees for duplicate keys?

Reply:
It will always retain the last-seen value from the `list`--i.e. the value whose key is at the greatest index. I am putting up another PR with `unnest` for map so will include this note in the docs in that PR to save a CI run.